packages/lmi/src/lmi/cost_tracker.py

```diff
@@ -26,6 +26,13 @@ def add_callback(self, callback: Callable[[LLMResponse], Awaitable]) -> None:
         self._callbacks.append(callback)
 
     async def record(self, response: LLMResponse) -> None:
+        # Only record on responses with usage information (final chunk in streaming)
+        # We check for usage presence rather than cost > 0 because:
+        # - Free models, unknown models, or custom pricing can have cost = 0
+        # - We still want to fire callbacks for these to maintain visibility
+        if not getattr(response, "usage", None):
```

Comment:
```suggestion
        if not hasattr(response, "usage"):
```

This may be better because it avoids a truthy check on `.usage` when present. Also, feel free to ignore this one

Reply:
it is because of this option

stream_options = {
            "include_usage": True,
} 
...
stream_completions = await track_costs_iter(self.router.acompletion)(
            self.name,
            prompts,
            stream=True,
            stream_options=stream_options,
            **kwargs,
        )

when it is true you get usage: null for every intermediate chunk

Reply:
Ah I see, then yeah we nees the truthy check. Thanks for answering